lib/rules/no-restricted-imports.js

```diff
@@ -678,6 +780,20 @@ module.exports = {
 				}
 			},
 			ExportAllDeclaration: checkNode,
+			// Add support for TypeScript import equals declarations
+			TSImportEqualsDeclaration(node) {
+				if (node.moduleReference.type === "TSExternalModuleReference") {
```

Comment:
[Testing] Removing this `if` doesn't fail any unit tests. This can be hit for namespace imports: `import B = A.B;`.